docs/components/copy.md

```diff
@@ -0,0 +1,183 @@
+# Copy
+
+KCopy is a component that displays provided text and allows a user to copy it to their clipboard.
+
+<KCopy :text="text" />
+
+```html
+<KCopy :text="text" />
+```
+
+## Props
+
+### text
+
+The copyable text.
+
+### badge
+
+Whether or not to display as a badge. Defaults to `false`.
+
+<KCopy
+  badge
+  truncate
+  :text="text"
+/>
+
+```html
+<KCopy
+  badge
+  truncate
+  :text="text"
+/>
+```
+
+### badgeText
+
+Text displayed before the copyable text when `badge` is true.
+
+<KCopy
+  badge
+  badge-text="Id"
+  :text="text"
+/>
+
+```html
+<KCopy
+  badge
+  badge-text="Id"
+  :text="text"
+/>
+```
+
+### truncate
+
+Whether or not the text should be truncated. Defaults to `false`.
+
+<KCopy
+  truncate
+  :text="text"
+/>
+
+```html
+<KCopy
+  truncate
+  :text="text"
+/>
+```
+
+### truncationLimit
+
+Number of characters to truncate at. Defaults to `8`.
+
+<KCopy
+  truncate
+  :text="text"
+  :truncation-limit="15"
+/>
+
+```html
+<KCopy
+  truncate
+  :text="text"
+  :truncation-limit="15"
+/>
+```
+
+### copyTooltip
+
+Tooltip text displayed on hover copy button. 
+If the `badgeText` prop has a value, then the copy tooltip text is `Copy {badgeText}`; otherwise, `Copy`.
+
+<KCopy
+  :text="text"
+  badge 
+  badgeText="ID:" 
+/>
+
+```html
+<KCopy
+  :text="text"
+  badge 
+  badgeText="ID:" 
+/>
+```
+
+### textTooltip
+
+Tooltip text displayed on hover over the `text`.
+
+<KCopy
+  truncate
+  :text="text"
+  text-tooltip="Hello!"
+/>
+
+```html
+<KCopy
+  truncate
+  :text="text"
+  text-tooltip="Hello!"
+/>
+```
+
+### successTooltip
+
+Tooltip text displayed on successful copy. Defaults to `Copied!`.
+
+### monospace
+
+An indicator of whether the copyable text has `JetBrains Mono` font or not. Defaults to `true`, but `false` if `badge` is `true`.
+
+### format
+
+Determines the display format of the copyable text. The component can take the following `format` values:
+
+- `default`: displays regular uuid
+- `hidden`: displays just a copy button without text
+- `redacted`: displays `*****`
+- `deleted`: displays `*<first-5-chars-of-uuid>`
+
+<KCopy
+  format="hidden"
+  :text="text"
+/>
+
+```html
+<KCopy
+  format="hidden"
+  :text="text"
+/>
+```
+
+<KCopy
+  format="redacted"
+  :text="text"
+/>
+
+```html
+<KCopy
+  format="redacted"
+  :text="text"
+/>
+```
+
+<KCopy
+  format="deleted"
+  :text="text"
+/>
+
+```html
+<KCopy
+  format="deleted"
+  :text="text"
+/>
+```
```

Comment:
suggestion: Can you add a header over each example so you can tell which is which?

Reply:
Does this look ok? 
<img width="768" alt="image 1646" src="https://github.com/Kong/kongponents/assets/87779967/e43218fa-a14d-4335-b78b-f2838d0b1b3e">

Reply:
nit: I'd just use h4/h5 tags, e.g.

`#### deleted` or h5 depending on where you are in the doc

Reply:
updated
<img width="1198" alt="image 1648" src="https://github.com/Kong/kongponents/assets/87779967/883e3b2a-d559-4ceb-a66a-a3045f600821">